test/context.ts

```diff
@@ -110,7 +110,7 @@ test('Get single user', async() => {
 
 test('Remove many users (returning ALL_OLD)', async() => {
     //  PK comes from context
-    let removed = await User.remove({}, {many: true})
+    let removed: any = await User.remove({}, {many: true})
     expect(removed).toHaveLength(3)
```

Comment:
Why is the any required here?

Reply:
again here its making an issue visible because the model is inferred. It seems that based on the model type its only expecting the entity returned but here we using an array. So the many is not recognised. We can explore to see if we can get this working in typescript but this is issue was not caused by the new code

Reply:
Yes, we should update the remove() signature to allow Array<T>

Reply:
ok so this should be in a new PR right?

Reply:
Yes, that would be best. 

The key open question for me is the backwards compatibility. i.e.

>Given this is breaking compatibility (table constructor), we'll need to defer to our 3.x branch which we plan to explore converting the source to TS. What are your thoughts?

Reply:
this already works like this though right? the only reason it's visible now is because models are inferred it is always typed. So if someone in the current version created a typed model they would see the same issue no?

Reply:
I may be getting confused.

Can you give me a little sample of exactly how the new TableConstructorParmams works?

Before

    type TableConstructorParams = {

Proposed

    type TableConstructorParams<Schema extends OneSchema> = {

Reply:
This is also new to me. I followed what was done on the previous PR to get it to work. So the goal was to get the models that are passed in the schema. To get access to this schema we pass an inferred type from table class to the constructor. The inferred schema type can then be used to expose the models to the model method. So the schema here is just a reference that gets populated from the constructor. Its not an expected type that needs to be explicitly passed.